src/Hl7.Fhir.Shims.Base/Specification/Terminology/CodeSystemFilterProcessor.cs

```diff
@@ -0,0 +1,167 @@
+﻿using Hl7.Fhir.Model;
+using Hl7.Fhir.Specification.Source;
+using Hl7.Fhir.Utility;
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using T = System.Threading.Tasks;
+
+#nullable enable
+
+namespace Hl7.Fhir.Specification.Terminology
+{
+
+    internal static class CodeSystemFilterProcessor
+    {
+        private static readonly ReadOnlyCollection<FilterOperator?> SUPPORTEDFILTERS = new(new List<FilterOperator?> { FilterOperator.IsA });
+        private const string SUBSUMEDBYCODE = "subsumedBy";
+
+        /// <summary>
+        /// Retrieve codes from a CodeSystem resource bases on one or multiple filters.
+        /// </summary>
+        /// <param name="codeSystemUri">Uri of the CodeSystem</param>
+        /// <param name="filters">Filters to be applied</param>
+        /// <param name="settings">ValueSetExpanderSettings </param>
+        /// <returns></returns>
+        /// <exception cref="ValueSetExpansionTooComplexException">Thrown when a filter is applied that is not supported (yet)</exception>
+        /// <exception cref="CodeSystemUnknownException">Thrown when no resource resolver was set in ValueSetExpanderSettings.ValueSetSource</exception>
+        /// <exception cref="CodeSystemUnknownException">Thrown when the requested CodeSystem can not be found by the resource resolver in ValueSetExpanderSettings</exception>
+        internal async static T.Task<IEnumerable<ValueSet.ContainsComponent>> FilterConceptsFromCodeSystem(string codeSystemUri, List<ValueSet.FilterComponent> filters, ValueSetExpanderSettings settings)
+        {
+            if (codeSystemUri == "http://snomed.info/sct" || codeSystemUri == "http://loinc.org")
+            {
+                throw new ValueSetExpansionTooComplexException($"Locally filter codes from Codesystem {codeSystemUri} is not supported");
+            }
+
+            if (filters.Any(f => !SUPPORTEDFILTERS.Contains(f.Op)))
```

Comment:
You could do this also in the switch expression on line 79. saves code & double checks.